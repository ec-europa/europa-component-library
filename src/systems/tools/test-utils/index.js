const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
