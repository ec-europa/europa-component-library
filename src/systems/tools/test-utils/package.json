{
  "name": "@ecl-twig/test-utils",
  "version": "2.31.1",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
