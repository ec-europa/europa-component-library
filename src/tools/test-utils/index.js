const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
