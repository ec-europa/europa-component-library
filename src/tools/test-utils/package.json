{
  "name": "@ecl/test-utils",
  "version": "3.4.2",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
