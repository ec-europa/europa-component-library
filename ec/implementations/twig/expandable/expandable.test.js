import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

import demoData from './demo/data';

// Add SVG icon path.
demoData.button.icon.path = 'example';

describe('EC - Expandable', () => {
  const template = '@ecl-twig/ec-component-expandable/ecl-expandable.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Collapsed', () => {
    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(demoData)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(demoData, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(demoData, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });

    test('with missing input data and debug enbaled returns the right warning message', () => {
      expect.assertions(1);

      const dataCompliance = { ...demoData, _compliance_: true };
      dataCompliance.label_expanded = '';

      return expect(render(dataCompliance)).resolves.toMatchSnapshot();
    });
  });

  describe('Expanded', () => {
    const expanded = { ...demoData, expanded: true };

    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(expanded)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(expanded, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(expanded, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });
});
