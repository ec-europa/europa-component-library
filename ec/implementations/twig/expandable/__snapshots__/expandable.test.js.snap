// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Expandable Collapsed renders correctly 1`] = `
<test>
  <div
    class="ecl-expandable"
    data-ecl-auto-init="Expandable"
    data-ecl-expandable="true"
  >
    <button
      aria-controls="expandable-example-content"
      aria-expanded="false"
      class="ecl-button ecl-button--secondary ecl-expandable__toggle"
      data-ecl-expandable-toggle=""
      data-ecl-label-collapsed="Collapsed button"
      data-ecl-label-expanded="Expanded button"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Collapsed button
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--fluid ecl-icon--rotate-180 ecl-button__icon ecl-button__icon--after"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="example#ui--corner-arrow"
          />
        </svg>
      </span>
    </button>
    <div
      class="ecl-expandable__content"
      hidden=""
      id="expandable-example-content"
    >
      <p
        class="ecl-u-type-paragraph-m"
      >
        The EU is building an energy union that make sures Europe’s energy supply is safe, viable and accessible to all. In doing so, it can boost the economy and attract investments that can create new jobs opportunities.
      </p>
    </div>
  </div>
</test>
`;

exports[`EC - Expandable Collapsed renders correctly with extra attributes 1`] = `
<test>
  <div
    class="ecl-expandable"
    data-ecl-auto-init="Expandable"
    data-ecl-expandable="true"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <button
      aria-controls="expandable-example-content"
      aria-expanded="false"
      class="ecl-button ecl-button--secondary ecl-expandable__toggle"
      data-ecl-expandable-toggle=""
      data-ecl-label-collapsed="Collapsed button"
      data-ecl-label-expanded="Expanded button"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Collapsed button
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--fluid ecl-icon--rotate-180 ecl-button__icon ecl-button__icon--after"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="example#ui--corner-arrow"
          />
        </svg>
      </span>
    </button>
    <div
      class="ecl-expandable__content"
      hidden=""
      id="expandable-example-content"
    >
      <p
        class="ecl-u-type-paragraph-m"
      >
        The EU is building an energy union that make sures Europe’s energy supply is safe, viable and accessible to all. In doing so, it can boost the economy and attract investments that can create new jobs opportunities.
      </p>
    </div>
  </div>
</test>
`;

exports[`EC - Expandable Collapsed renders correctly with extra class names 1`] = `
<test>
  <div
    class="ecl-expandable custom-class custom-class--test"
    data-ecl-auto-init="Expandable"
    data-ecl-expandable="true"
  >
    <button
      aria-controls="expandable-example-content"
      aria-expanded="false"
      class="ecl-button ecl-button--secondary ecl-expandable__toggle"
      data-ecl-expandable-toggle=""
      data-ecl-label-collapsed="Collapsed button"
      data-ecl-label-expanded="Expanded button"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Collapsed button
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--fluid ecl-icon--rotate-180 ecl-button__icon ecl-button__icon--after"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="example#ui--corner-arrow"
          />
        </svg>
      </span>
    </button>
    <div
      class="ecl-expandable__content"
      hidden=""
      id="expandable-example-content"
    >
      <p
        class="ecl-u-type-paragraph-m"
      >
        The EU is building an energy union that make sures Europe’s energy supply is safe, viable and accessible to all. In doing so, it can boost the economy and attract investments that can create new jobs opportunities.
      </p>
    </div>
  </div>
</test>
`;

exports[`EC - Expandable Collapsed with missing input data and debug enbaled returns the right warning message 1`] = `
<test>
  <div
    class="ecl-expandable"
    data-ecl-auto-init="Expandable"
    data-ecl-expandable="true"
  >
    <button
      aria-controls="expandable-example-content"
      aria-expanded="false"
      class="ecl-button ecl-button--secondary ecl-expandable__toggle"
      data-ecl-expandable-toggle=""
      data-ecl-label-collapsed="Collapsed button"
      data-ecl-label-expanded=""
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Collapsed button
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--fluid ecl-icon--rotate-180 ecl-button__icon ecl-button__icon--after"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="example#ui--corner-arrow"
          />
        </svg>
        <span
          class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
          title="[icon] - ecl compliant"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="example#notifications--success"
            />
          </svg>
        </span>
      </span>
    </button>
    <span
      class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
      title="[button--secondary] - ecl compliant"
    >
      <svg
        aria-hidden="true"
        class="ecl-icon ecl-icon--xs"
        focusable="false"
      >
        <use
          xlink:href="example#notifications--success"
        />
      </svg>
    </span>
    <div
      class="ecl-expandable__content"
      hidden=""
      id="expandable-example-content"
    >
      <p
        class="ecl-u-type-paragraph-m"
      >
        The EU is building an energy union that make sures Europe’s energy supply is safe, viable and accessible to all. In doing so, it can boost the economy and attract investments that can create new jobs opportunities.
      </p>
    </div>
  </div>
  <!-- ecl2-compliance: expandable -->
  <div
    class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-red ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
  >
    <p
      class="ecl-u-type-color-grey ecl-u-ml-m ecl-u-type-xs"
    >
      [expandable] -  {{ label_expanded }} property needs to be set and filled in.
    </p>
  </div>
</test>
`;

exports[`EC - Expandable Expanded renders correctly 1`] = `
<test>
  <div
    class="ecl-expandable"
    data-ecl-auto-init="Expandable"
    data-ecl-expandable="true"
  >
    <button
      aria-controls="expandable-example-content"
      aria-expanded="false"
      class="ecl-button ecl-button--secondary ecl-expandable__toggle"
      data-ecl-expandable-toggle=""
      data-ecl-label-collapsed="Collapsed button"
      data-ecl-label-expanded="Expanded button"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Collapsed button
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--fluid ecl-icon--rotate-180 ecl-button__icon ecl-button__icon--after"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="example#ui--corner-arrow"
          />
        </svg>
      </span>
    </button>
    <div
      class="ecl-expandable__content"
      hidden=""
      id="expandable-example-content"
    >
      <p
        class="ecl-u-type-paragraph-m"
      >
        The EU is building an energy union that make sures Europe’s energy supply is safe, viable and accessible to all. In doing so, it can boost the economy and attract investments that can create new jobs opportunities.
      </p>
    </div>
  </div>
</test>
`;

exports[`EC - Expandable Expanded renders correctly with extra attributes 1`] = `
<test>
  <div
    class="ecl-expandable"
    data-ecl-auto-init="Expandable"
    data-ecl-expandable="true"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <button
      aria-controls="expandable-example-content"
      aria-expanded="false"
      class="ecl-button ecl-button--secondary ecl-expandable__toggle"
      data-ecl-expandable-toggle=""
      data-ecl-label-collapsed="Collapsed button"
      data-ecl-label-expanded="Expanded button"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Collapsed button
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--fluid ecl-icon--rotate-180 ecl-button__icon ecl-button__icon--after"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="example#ui--corner-arrow"
          />
        </svg>
      </span>
    </button>
    <div
      class="ecl-expandable__content"
      hidden=""
      id="expandable-example-content"
    >
      <p
        class="ecl-u-type-paragraph-m"
      >
        The EU is building an energy union that make sures Europe’s energy supply is safe, viable and accessible to all. In doing so, it can boost the economy and attract investments that can create new jobs opportunities.
      </p>
    </div>
  </div>
</test>
`;

exports[`EC - Expandable Expanded renders correctly with extra class names 1`] = `
<test>
  <div
    class="ecl-expandable custom-class custom-class--test"
    data-ecl-auto-init="Expandable"
    data-ecl-expandable="true"
  >
    <button
      aria-controls="expandable-example-content"
      aria-expanded="false"
      class="ecl-button ecl-button--secondary ecl-expandable__toggle"
      data-ecl-expandable-toggle=""
      data-ecl-label-collapsed="Collapsed button"
      data-ecl-label-expanded="Expanded button"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Collapsed button
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--fluid ecl-icon--rotate-180 ecl-button__icon ecl-button__icon--after"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="example#ui--corner-arrow"
          />
        </svg>
      </span>
    </button>
    <div
      class="ecl-expandable__content"
      hidden=""
      id="expandable-example-content"
    >
      <p
        class="ecl-u-type-paragraph-m"
      >
        The EU is building an energy union that make sures Europe’s energy supply is safe, viable and accessible to all. In doing so, it can boost the economy and attract investments that can create new jobs opportunities.
      </p>
    </div>
  </div>
</test>
`;
